### DISCUSSION TITLE
Is there a way to keep track of socket connections?

### DISCUSSION THREAD

**Discussion by ddevienne** [uNetworking/uWebSockets#1679] [Q&A]
I'd like to know when client socket connections come (connect) and go (disconnect / close).
With Keep-Alive, many HTTP requests can be on the same connections.
Is there a way in uWebSockets or the underlying uSockets to be notified of those?
That way I can know how many _concurrent_ connections are serviced, and for how long.

Might also be useful to close idle connections (although there might be a way in uSockets to achieve that?)
or refuse connections from hosts/addresses with too many errors from earlier connections.

PS: Note that I wrote [_concurrent_ and not _parallel_][1]. 

[1]: https://go.dev/blog/waza-talk

**Comment by uNetworkingAB** [comment]
App.filter attaches a function that gets differences in number of connections

**Reply by uNetworkingAB** [reply]
You can also filter connections in that callback, if they are from some IP or whatever
